plugins/@grouparoo/snowflake/src/lib/export/exportRecords.ts

```diff
@@ -75,63 +75,37 @@ const updateByDestinationIds: BatchMethodUpdateByDestinationIds = async ({
 }) => {
   const table: string = config.destinationOptions.table?.toString();
   const primaryKey: string = config.destinationOptions.primaryKey?.toString();
-  const payloads: any[] = [];
-  let columns: any = [];
-  let columnsToSet: string = "";
 
   const { primaryKeysToDelete, currentPrimaryKeyValues } =
     await treatCollisions(client, table, primaryKey, users);
-
-  for (const record of users) {
-    const payload = buildPayload(record);
-    if (columns.length === 0 || columns.length < Object.keys(payload).length) {
-      columns = Object.keys(payload);
-      columnsToSet = Object.keys(payload)
-        .map((entry) => `${entry} = newRecord.${entry}`)
-        .join(", ");
-    }
-    payloads.push(payload);
-  }
-
   if (primaryKeysToDelete.length > 0) {
     // delete/orphan old primary keys
     await deleteOrClearGroups(client, config, primaryKeysToDelete);
   }
-
-  let values: any = [];
-  for (let payload of payloads) {
-    payload = normalizePayload(payload, columns);
-    payload["currentPrimaryKeyValue"] =
-      currentPrimaryKeyValues[payload[primaryKey]];
-    const cleanedValues = formatPayloadValues(Object.values(payload));
-    values.push(`(${cleanedValues})`);
-  }
-  values = values.join(", ");
+  const { columnsToSet, columns, values } = buildColumnsAndValues(
+    users,
+    primaryKey,
+    currentPrimaryKeyValues
+  );
   columns.push("CURRENT_PRIMARY_KEY");
-  columns = columns.join(", ");
-  const query = `UPDATE ${table} SET ${columnsToSet} FROM (VALUES ${values}) newRecord(${columns}) WHERE ${table}.${primaryKey} = newRecord.CURRENT_PRIMARY_KEY`;
-  await client.execute(query);
+  const singleValuePlaceholder = Array.from(
+    { length: columns.length },
+    () => "?"
+  );
+  const valuesPlaceholders = Array.from(
+    { length: values.length },
+    () => `(${singleValuePlaceholder})`
```

Comment:
I think i'd feel better if this was like `singleValuePlaceholder.join(", ")` instead of relying on the `toString` behavior.